BSP/Inc/BSP_Timer.h

```diff
@@ -6,18 +6,26 @@
 #include "common.h"
 
 /**
- * @brief   Initialize the timer for time measurements.
+ * @brief   Initialize timers
  * @param   None
  * @return  None
  */
 void BSP_Timer_Init(void);
 
 /**
- * @brief   Starts the timer.
+ * @brief   Starts a one shot timer to execute a callback after a certain time
+ * 
+ * @param delay_us one shot time in microseconds
+ * @param callback callback to execute after `delay_us` time
+ */
```

Comment:
Add a note saying that the delay_us should be less than the tick count for the RTOS. If it is larger and this is called from multiple threads, then we have a race condition.

Reply:
and any other issues you could think of.